src/aind_metadata_service/slims/procedures/mapping.py

```diff
@@ -0,0 +1,216 @@
+                    self._map_antibody(wash) if wash else []
+                ),  # Antibody mapping
+            )
+            procedures.append(spec)
+        return procedures
+
+    @staticmethod
+    def _map_antibody(wash: SlimsWash) -> Optional[Antibody]:
+        """Map antibody reagent from wash"""
+        if not hasattr(wash, "wash_step") or wash.wash_step is None:
+            return None
+        wash_name = getattr(wash.wash_step, "wash_name", None)
+        if wash_name == SlimsWashNames.PRIMARY_ANTIBODY_WASH.value:
+            label = ImmunolabelClass.PRIMARY
+        elif wash_name == SlimsWashNames.SECONDARY_ANTIBODY_WASH.value:
+            label = ImmunolabelClass.SECONDARY
+        else:
+            return None
+        mass = getattr(wash.wash_step, "mass", None)
+        return Antibody.model_construct(
+            immunolabel_class=label,
+            mass=mass,
+        )
+
+    def _map_reagents(self, washes: List[SlimsWash]) -> List[Reagent]:
+        """Map info from washes"""
+        reagents = []
+        for wash in washes:
+            for reagent in wash.reagents:
+                reagent_model = Reagent.model_construct(
+                    name=getattr(reagent.details, "name", None),
+                    source=(
+                        self._map_source(reagent.source.name)
+                        if getattr(reagent.source, "name", None)
+                        else None
+                    ),
+                    lot_number=getattr(reagent.content, "lot_number", None),
+                )
+                reagents.append(reagent_model)
+        return reagents
+
+    @staticmethod
+    def _map_source(source_name: str) -> Optional[Organization]:
+        """Map source name to source type"""
+        return Organization.from_name(source_name)
+
+    @staticmethod
+    def _map_procedure_type(
+        block_name: str,
+    ) -> Optional[SpecimenProcedureType]:
+        """Map procedure type from experiment template name"""
+        if (
+            block_name == SlimsExperimentTemplateNames.SMARTSPIM_LABELING
+            or block_name == SlimsExperimentTemplateNames.EXASPIM_LABELING
+        ):
+            return SpecimenProcedureType.IMMUNOLABELING
+        elif (
+            block_name == SlimsExperimentTemplateNames.SMARTSPIM_DELIPIDATION
+            or block_name == SlimsExperimentTemplateNames.EXASPIM_DELIPIDATION
+        ):
+            return SpecimenProcedureType.DELIPIDATION
+        elif block_name == SlimsExperimentTemplateNames.SMARTSPIM_RI_MATCHING:
+            return SpecimenProcedureType.REFRACTIVE_INDEX_MATCHING
+        elif block_name == SlimsExperimentTemplateNames.EXASPIM_GELATION:
+            return SpecimenProcedureType.GELATION
+        elif block_name == SlimsExperimentTemplateNames.EXASPIM_EXPANSION:
+            return SpecimenProcedureType.EXPANSION
+        else:
+            return None
+
+    @staticmethod
+    def _map_experimenters(washes: List[SlimsWash]):
```

Comment:
@saskiad Currently, I am mapping the experimenter name from the SLIMS account who entered the info in the SPIM Histology workflow. Is this alright, or would you rather I leave this empty for now (and request that we add an entry field for experimenters)

Reply:
I think this is okay. We should ask Abbie and Clark if this makes sense for how SLIMS is being used or if we should add a person's name field.

Reply:
Abbie said this is fine